pandas/_libs/missing.pyx

```diff
@@ -334,6 +334,22 @@ def isnaobj2d_old(arr: ndarray) -> ndarray:
     return result.view(np.bool_)
 
 
+@cython.wraparound(False)
```

Comment:
this is the same as: `cpdef ndarray[uint8_t] isnaobj(ndarray arr)` no?

Reply:
its might be a very tiny bit faster but likely not worth it

Reply:
The issue here was that we need to be able to explicitly check for `NA` (and not other missing values) to maintain the existing behavior

Reply:
we actually test this? i think we accept multiple types of Nulls here (e.g. None is accepted as well as np.nan for strings). I would just use our existing routine unless you have tests that show this is not sufficient.

Reply:
This is not tested (hence the regression :)

On 1.2.x the behavior is
```
In [1]: import pandas as pd
  
In [2]: import numpy as np

In [3]: ser = pd.Series([1, 1, pd.NA], dtype="Int64")

In [4]: ser.isin([pd.NA])
Out[4]: 
0    False
1    False
2     True
dtype: bool

In [5]: ser.isin([np.nan])
Out[5]: 
0    False
1    False
2    False
dtype: bool
```

This `isin` behavior singling out `pd.NA` also matches strings:
```
In [7]: ser = pd.Series(["a", "b", None], dtype="string")

In [8]: ser
Out[8]: 
0       a
1       b
2    <NA>
dtype: string

In [9]: ser.isin([np.nan])
Out[9]: 
0    False
1    False
2    False
dtype: bool

In [10]: ser.isin([pd.NA])
Out[10]: 
0    False
1    False
2     True
dtype: bool
```

So this matches existing behavior and how `StringDtype` handles `NA` for `isin`, even if the behavior itself may be questionable

Reply:
sure. this should match anything we allow for nulls in the `_from_sequence` which is what `isnaobj` does

Reply:
So just to make sure we're on the same page, the following 1.2.x behavior can be considered a bug:
```
In [4]: ser
Out[4]: 
0       1
1    <NA>
dtype: Int64

In [5]: ser.isin([np.nan])
Out[5]: 
0    False
1    False   # Should be True, matching ser.isin([pd.NA])?
dtype: bool
```
And if so, this change is ok to include in a 1.3.x backport?

Reply:
is this a change or how it was?

Reply:
The current state of the pr maintains the 1.2.x behavior shown in the `ipython` blocks earlier in this thread. Accepting multiple nulls instead of just NA would change 1.2.x behavior from 
```
In [4]: ser
Out[4]: 
0   <NA>
dtype: Int64

In [5]: ser.isin([np.nan])
Out[5]: 
0    False   
dtype: bool
```
to 
```
In [4]: ser
Out[4]: 
0   <NA>
dtype: Int64

In [5]: ser.isin([np.nan])
Out[5]: 
0    True   
dtype: bool
```

Reply:
did you add an explict test for this? i think the existing behavior is wrong; here we are treating nans that accepted in construction *differently* which is a no-no.

cc @jbrockmendel @jorisvandenbossche

Reply:
> did you add an explict test for this?
The added test has parameterizations which hit this explicitly.

Reply:
> also -1 on the expansion of yet another null checker. if we want to keep the existing then do this via a flag to isnaobj.

If we're uncertain about this behavior, could also just turn the added cython routine into something similar done in python space just to fix the regression. Then that could easily be removed for 1.4 if we decide to change the behavior.

Reply:
> sure. this should match anything we allow for nulls in the _from_sequence which is what isnaobj does

i havent looked at this closely, but the canonical behavior belongs in is_valid_na_for_dtype

Reply:
I think we're going to at some point need a vectorized is_matching_na, which this would be a special case of

Reply:
> i havent looked at this closely, but the canonical behavior belongs in is_valid_na_for_dtype

Not sure if along the lines of what you were thinking, but pushed a different solution just checking for presence of a valid missing value using `dtype.na_value`